src/Core/Grid/DataProvider/DoctrineGridDataProvider.php

```diff
@@ -58,6 +75,12 @@ public function getData(SearchCriteriaInterface $searchCriteria)
         $searchQueryBuilder = $this->gridQueryBuilder->getSearchQueryBuilder($searchCriteria);
         $countQueryBuilder = $this->gridQueryBuilder->getCountQueryBuilder($searchCriteria);
 
+        $this->hookDispatcher->dispatchForParameters('modifyGridQueryBuilder', [
```

Comment:
I think hooks are typically prepended with "action", aren't they?

https://devdocs.prestashop.com/1.7/modules/concepts/hooks/#naming-scheme

Reply:
well, i guess you are right, although there are some hooks that does not respect this rule. :/ would `actionModifyGridQueryBuilder` be better?

Regarding you comment about applicators, i think that this hook will not be needed anymore, as we could allow attaching custom applicators instead?

Reply:
we need a hook to be able to attach custom applicators instead then :-)

And this should remains easy to use it